RELEASE.md

```diff
@@ -0,0 +1,58 @@
+# Gophercloud release
+
+## Contributions
+
+### The semver label
+
+Gophercloud follows [semver](https://semver.org/).
+
+Each Pull request must have a label indicating its impact on the API:
+* `semver:patch` for changes that don't impact the API
+* `semver:minor` for changes that impact the API in a backwards-compatible fashion
+* `semver:major` for changes that introduce a breaking change in the API
+
+Automation prevents merges if the label is not present.
+
+### Metadata
+
+The release notes for a given release are generated based on the PR title and its milestone:
+* make sure that the PR title is descriptive
+* add a milestone based on the semver label: x++ if major, y++ if minor, z++ if patch.
+
+## Release of a new version
+
+### Step 1: Check the metadata
+
+Check that all pull requests merged since the last release have the right milestone.
+
+### Step 2: Release notes and version string
+
+Once all PRs have a sensible title and are added to the right milestone, generate the release notes with the [`gh`](https://github.com/cli/cli) tool:
+```shell
+gh pr list \
```

Comment:
For later, we could think of a script to help prepare for the release:
- fetch all PRs that have merged since last release
- propose a new version based on the semver label of these PRs
- check that all relevant PRs (the ones that need a line in the changelog) have the appropriate milestone and set it if not
- generate changelog

Reply:
absolutely. Also, once a script is in place we might not even strictly need milestones any more.

Reply:
absolutely. Also, once a script is in place we might not even strictly need milestones any more.